### PR TITLE
fix(smart-contracts): deploy and verify batch contract

### PR DESCRIPTION
**Pull request by olivier7delf**
Batch contracts deployed and verified: bsc, avalanche, and Rinkeby (address updated)
Fix setup batch deployment
Fix gnosis WEB3_PROVIDER_URL
Refacto networkRPC utils

### CODE REVIEW

File: packages/smart-contracts/scripts-create2/contract-setup/setupBatchPayments.ts [hunk 1]
```diff
@@ -35,10 +35,11 @@ export const setupBatchPayments = async (
       const adminNonce = await signer.getTransactionCount();
       const gasPrice = await provider.getGasPrice();
 
+      // start to add 0 to the adminNonce, increase gasPrice if needed
```
Comment (bertux) [inline packages/smart-contracts/scripts-create2/contract-setup/setupBatchPayments.ts:38]:
```suggestion
      // start from the adminNonce, increase gasPrice if needed
```

File: packages/smart-contracts/scripts-create2/contract-setup/setupBatchPayments.ts [hunk 2]
```diff
@@ -35,10 +35,11 @@ export const setupBatchPayments = async (
...
       await Promise.all([
-        updateBatchPaymentFees(batchPaymentConnected, adminNonce + 3, gasPrice),
-        updatePaymentErc20FeeProxy(batchPaymentConnected, network, adminNonce + 4, gasPrice),
-        updatePaymentEthFeeProxy(batchPaymentConnected, network, adminNonce + 5, gasPrice),
+        updateBatchPaymentFees(batchPaymentConnected, adminNonce + 0, gasPrice),
```
Comment (bertux) [inline packages/smart-contracts/scripts-create2/contract-setup/setupBatchPayments.ts:40]:
```suggestion
        updateBatchPaymentFees(batchPaymentConnected, adminNonce, gasPrice),
```

**Review by bertux — APPROVED** [review]

File: packages/utils/src/providers.ts [hunk 3]
```diff
@@ -36,7 +36,7 @@ const networkRpcs: Record<string, string> = {
   fuse: 'https://rpc.fuse.io',
   bsctest: 'https://data-seed-prebsc-1-s1.binance.org:8545',
   bsc: 'https://bsc-dataseed1.binance.org/',
-  xdai: 'https://rpc.xdaichain.com/',
+  xdai: 'https://gnosischain-rpc.gateway.pokt.network/',
```
Comment (bertux) [inline packages/utils/src/providers.ts:39]:
```suggestion
  xdai: 'https://gnosischain-rpc.gateway.pokt.network/',
  celo: 'https://forno.celo.org',
```

**Review by bertux — CHANGES_REQUESTED** [review]

**Review by bertux — APPROVED** [review]

File: packages/smart-contracts/hardhat.config.ts [hunk 4]
```diff
@@ -66,52 +67,52 @@ export default {
       accounts,
     },
     matic: {
-      url: process.env.WEB3_PROVIDER_URL || 'https://polygon-rpc.com',
+      url: process.env.WEB3_PROVIDER_URL || utils.networkRpcs.matic,
```
Comment (yomarion) [inline packages/smart-contracts/hardhat.config.ts:70]: Would be nice to factorize a bit:
```suggestion
// Once and for all
      const url = (network) => process.env.WEB3_PROVIDER_URL || utils.networkRpcs[network]
// For every network
      url: url(matic)
```

File: packages/smart-contracts/scripts-create2/contract-setup/adminTasks.ts [hunk 5]
```diff
@@ -88,6 +90,8 @@ export const updatePaymentEthFeeProxy = async (
   const ethereumFeeProxyAddress = ethereumFeeProxy.getAddress(network);
   const currentAddress = await contract.paymentEthFeeProxy();
   if (currentAddress !== ethereumFeeProxyAddress) {
+    console.log('currentAddress', currentAddress);
+    console.log('ethereumFeeProxyAddress', ethereumFeeProxyAddress);
```
Comment (yomarion) [inline packages/smart-contracts/scripts-create2/contract-setup/adminTasks.ts:94]: 4 console logs

**Review by yomarion — APPROVED** [review]

**Pull request merged**